runhouse/resources/secrets/provider_secrets/provider_secret.py

```diff
@@ -75,13 +75,10 @@ def from_config(config: dict, dryrun: bool = False):
         """Create a ProviderSecret object from a config dictionary."""
         return ProviderSecret(**config, dryrun=dryrun)
 
-    def save(self, values: bool = True, headers: str = rns_client.request_headers):
-        if not self.name:
-            self.name = self.provider
-        super().save(values=values, headers=headers)
-
-    def delete(self, headers: str = rns_client.request_headers, contents: bool = False):
-        """Delete the secret config from Vault/local. Optionally also delete contents of secret file or env vars."""
+    def delete(self, headers: Optional[str] = None, contents: bool = False):
+        """Delete the secret config from Den and from Vault/local. Optionally also delete contents of secret file
+        or env vars."""
+        headers = headers or rns_client.request_headers
```

Comment:
what's the reason for separating out the definition this way instead of just having `rns_client.request_headers` be the default value?